ls1intum/Artemis
# [Exam] Enhancements for student exam generation

### Checklist

### Description

- [x] Show confirmation dialog before the generation of student exams, if some student exams already exist
- [x] Added button, which only creates individual exams for students who don't have one yet
- [x] Added status indicator for student exams, which shows whether individual exams exist for all registered students. Thanks @sascha11110 for helping with the styling.
- [x] Extended delete dialog of students for an exam. The associated student exam will also be deleted. Additionally the instructor has the option to also delete associated participations and submissions.

### Steps for Testing


1. Create an exam and properly set it up with some exercises
2. Register students for the exam
3. Go to the student exam page
--> Warning should be shown that not all registered students have an exam
4. Click on 'Generate individual exams'
--> No dialog shoud open. Right amount of exams should be created. Status should turn green. Check the individual exams
5. Click again on 'Generate individual exams'
--> Dialog should open, prompting for a confirmation
6. Add more students to the exam
7. Move back to the student exam page
--> Warning state should be displayed
8. Click on 'Generate missing individual exams'
--> Right amount of individual exams should be created. State should turn green. Check the newly created individual exams
9. Click again on 'Generate missing individual exams'
--> 0 exams should be created
10. Create participations by clicking on 'Prepare exercise start'
11. Unregister a student from the exam. Select the option with participations.
12. Move back to the student exam page
--> Exam of the student should be deleted. State should be green.

### Screenshots


![generateIndividualExam](https://user-images.githubusercontent.com/16407766/86293304-a4d31c80-bbf2-11ea-97eb-9b7b35e36c9e.PNG)
New generate student exam dialog, shown when some student exams already exist

![newDeleteStudentDialog](https://user-images.githubusercontent.com/16407766/86293307-a56bb300-bbf2-11ea-958a-de171be13e49.PNG)
Extended delete student from exam dialog

![studentExamsOk](https://user-images.githubusercontent.com/16407766/86293308-a6044980-bbf2-11ea-9640-8e3b7d1e4b90.PNG)
All students have an exam (or no registered users)

![studentExamWarn](https://user-images.githubusercontent.com/16407766/86293309-a6044980-bbf2-11ea-9e24-c1834e54ac64.png)
Not all students have an exam
